### PR TITLE
Removes already applied migrations

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

**Review by acatangiu — APPROVED** [review]
Looks good, but let's merge after we see them applied on-chain.

File: relay/polkadot/src/lib.rs [hunk 1]
```diff
@@ -1688,106 +1688,13 @@ pub type Migrations = migrations::Unreleased;
...
-
-	/// Upgrade Session keys to include BEEFY key.
-	/// When this is removed, should also remove `OldSessionKeys`.
-	pub struct UpgradeSessionKeys;
-	impl frame_support::traits::OnRuntimeUpgrade for UpgradeSessionKeys {
-		fn on_runtime_upgrade() -> Weight {
-			Session::upgrade_keys::<OldSessionKeys, _>(transform_session_keys);
-			Perbill::from_percent(50) * BlockWeights::get().max_block
-		}
-	}
```
Comment (acatangiu) [inline relay/polkadot/src/lib.rs:1745]: This migration was never released (and obviously never enacted on-chain) - don't remove it before we see it enacted.

**Review by acatangiu — CHANGES_REQUESTED** [review]
I saw comment https://github.com/polkadot-fellows/runtimes/issues/68#issuecomment-1849742955 where it says you want to merge this PR for pending `1.1.0` release, but some migrations here were never released (they were added after `1.0.0` release).

So we should not include/merge this PR (at least in its current form) before `1.1.0` release.

File: relay/polkadot/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply relay/polkadot/src/lib.rs:1745]: Haha, I didn't wanted to remove it :P I even left it in the list below :D Will readd the code, ty!

**Comment by acatangiu** [comment]
FYI: opened a PR for a doc with a quick guide that IMO we should follow to make our lives easier: https://github.com/polkadot-fellows/runtimes/pull/120

File: relay/polkadot/src/lib.rs [hunk 2]
```diff
@@ -1744,50 +1699,12 @@ pub mod migrations {
...
-		// Migrate parachain info format
-		paras_registrar::migration::VersionCheckedMigrateToV1<Runtime, ParachainsToUnlock>,
-
 		// Upgrade SessionKeys to include BEEFY key
 		UpgradeSessionKeys,
-
 		pallet_nomination_pools::migration::versioned_migrations::V5toV6<Runtime>,
 		pallet_nomination_pools::migration::versioned_migrations::V6ToV7<Runtime>,
-
-		runtime_parachains::scheduler::migration::v1::MigrateToV1<Runtime>
```
Comment (acatangiu) [inline relay/polkadot/src/lib.rs:1790]: not sure how to check if this was already run on-chain, but I see it was added here after the last release:
- active on-chain is spec `polkadot/1_000_001` which was [released](https://github.com/polkadot-fellows/runtimes/releases/tag/v1.0.1) on `22.10.2023`
- added in https://github.com/polkadot-fellows/runtimes/pull/75 merged a month later on `22.11.2023`

File: system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs [hunk 3]
```diff
@@ -909,11 +909,7 @@ pub type SignedExtra = (
 pub type UncheckedExtrinsic =
 	generic::UncheckedExtrinsic<Address, RuntimeCall, Signature, SignedExtra>;
 /// Migrations to apply on runtime upgrade.
-pub type Migrations = (
-	pallet_collator_selection::migration::v1::MigrateToV1<Runtime>,
-	pallet_multisig::migrations::v1::MigrateToV1<Runtime>,
-	InitStorageVersions,
```
Comment (acatangiu) [inline system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs:915]: maybe also remove `struct InitStorageVersions` defined and implemented below

(same for other system paras)

File: relay/polkadot/src/lib.rs (hunk 2 above)
Comment (bkchr) [reply relay/polkadot/src/lib.rs:1790]: This was added after the release, to switch back to the fixed upstream one. However, I had included the fixed upstream migration directly in this repo here: https://github.com/polkadot-fellows/runtimes/pull/26/files#diff-efa4caeb17487ecb13d8f5eb7863c3241d84afa2e73fbf25909a2ca89df0f362R1739

File: system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs (hunk 3 above)
Comment (bkchr) [reply system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs:915]: Done

File: system-parachains/asset-hubs/asset-hub-polkadot/src/lib.rs [hunk 4]
```diff
@@ -817,49 +817,14 @@ pub type UncheckedExtrinsic =
 	generic::UncheckedExtrinsic<Address, RuntimeCall, Signature, SignedExtra>;
 /// Migrations to apply on runtime upgrade.
 pub type Migrations =
-	(pallet_collator_selection::migration::v1::MigrateToV1<Runtime>, InitStorageVersions);
+	frame_support::migrations::VersionedMigration<0, 1, UniquesMigration, Uniques, RocksDbWeight>;
```
Comment (liamaharon) [inline system-parachains/asset-hubs/asset-hub-polkadot/src/lib.rs:820]: If the `InitStorageVersions` was executed, we need to reset the `Uniques` on-chain version back to 0 before running this.

Comment (bkchr) [reply system-parachains/asset-hubs/asset-hub-polkadot/src/lib.rs:820]: No this init was never executed on chain. This was the reason the CI was failing for this pr.

**Review by joepetrowski — APPROVED** [review]

File: system-parachains/asset-hubs/asset-hub-polkadot/src/lib.rs (hunk 4 above)
Comment (muharem) [reply system-parachains/asset-hubs/asset-hub-polkadot/src/lib.rs:820]: I wonder why it was not, I can see it here - https://github.com/polkadot-fellows/runtimes/blob/v1.0.0/system-parachains/asset-hubs/asset-hub-polkadot/src/lib.rs#L799

Comment (bkchr) [reply system-parachains/asset-hubs/asset-hub-polkadot/src/lib.rs:820]: Yes the migration was there. But the init of Uniques was added here: https://github.com/polkadot-fellows/runtimes/pull/87/files#diff-2dc481414ffabfb3b0b2c24f3c0cfc8a8b3ed6b56689a696d02be635795f5cb6R857

**Review by ggwpez — APPROVED** [review]
I dont know why we use `--disable-idempotency-checks` in the CI migration checks.  
I will try it in a MR to see if we can enable it now.

**Comment by bkchr** [comment]
> I dont know why we use `--disable-idempotency-checks` in the CI migration checks.

https://github.com/polkadot-fellows/runtimes/pull/75/files#diff-9c4bfd050ca12885b582fbfd04536460a7a4a0770824ee46694010b1e4e53287R74

**Comment by ggwpez** [comment]
> > I dont know why we use `--disable-idempotency-checks` in the CI migration checks.
> 
> https://github.com/polkadot-fellows/runtimes/pull/75/files#diff-9c4bfd050ca12885b582fbfd04536460a7a4a0770824ee46694010b1e4e53287R74

Yea I cannot read 🙈  removed in #133

**Comment by ggwpez** [comment]
@muharem its not counting your approval since your on-chain identity does not contain a github in the `additional` field.

**Review by sandreim — APPROVED** [review]

**Comment by sandreim** [comment]
> @muharem its not counting your approval since your on-chain identity does not contain a github in the `additional` field.

it's not counting mine as well and I have my github in the [additional](https://polkadot.statescan.io/#/accounts/13QdJvnJgfoitjrxESwrCWTaLMN8KvXxufDUucXM6EWGuxqh?tab=identity&page=1) field

File: relay/polkadot/src/lib.rs [hunk 5]
```diff
@@ -1744,50 +1699,12 @@ pub mod migrations {
...
-		frame_support::migrations::RemovePallet<PhragmenElectionPalletName, <Runtime as frame_system::Config>::DbWeight>,
-		frame_support::migrations::RemovePallet<TechnicalMembershipPalletName, <Runtime as frame_system::Config>::DbWeight>,
-		frame_support::migrations::RemovePallet<TipsPalletName, <Runtime as frame_system::Config>::DbWeight>,
-
-		parachains_configuration::migration::v9::MigrateToV9<Runtime>,
-		// Migrate parachain info format
-		paras_registrar::migration::VersionCheckedMigrateToV1<Runtime, ParachainsToUnlock>,
-
 		// Upgrade SessionKeys to include BEEFY key
 		UpgradeSessionKeys,
```
Comment (ordian) [inline relay/polkadot/src/lib.rs:1705]: is there an issue to track the removal of this in the next release so we don't forget?

**Review by ordian — APPROVED** [review]
Checked polkadot [migration list](https://github.com/polkadot-fellows/runtimes/blob/5d98c2efcdb61a90e5a325c80b6c5d2b2c1d6625/relay/polkadot/src/lib.rs#L1573-L1596) in 1.0.1 and it checks out.

**Comment by bkchr** [comment]
> > @muharem its not counting your approval since your on-chain identity does not contain a github in the `additional` field.
> 
> it's not counting mine as well and I have my github in the [additional](https://polkadot.statescan.io/#/accounts/13QdJvnJgfoitjrxESwrCWTaLMN8KvXxufDUucXM6EWGuxqh?tab=identity&page=1) field

The logs say something different:
```
ggwpez,acatangiu,kianenigma,shawntabrizi,gavofyork,davxy,eskimor,edwardmack,andresilva,pepyakin,rphmeier,bkchr,sandreim,tomaka,agryaznov,xlc,arkpar,ordian,KiChjang,noot,joepetrowski
Latest reviews are [{"user":"acatangiu","state":"APPROVED"},{"user":"joepetrowski","state":"APPROVED"},{"user":"muharem","state":"APPROVED"},{"user":"ggwpez","state":"APPROVED"},{"user":"sandreim","state":"APPROVED"},{"user":"ordian","state":"APPROVED"}]
```

File: relay/polkadot/src/lib.rs (hunk 5 above)
Comment (bkchr) [reply relay/polkadot/src/lib.rs:1705]: No we don't have an issue for this, but in the fellowship repo we can just remove always all the migrations.

**Pull request merged**